app/controllers/application_controller.rb

```diff
@@ -2,9 +2,126 @@
 
 class ApplicationController < Sinatra::Base
 
-  configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
-  end
+	configure do
+		set :public_folder, 'public'
+		set :views, 'app/views'
+		enable :sessions
+	end
 
-end
\ No newline at end of file
+	get '/' do
+		erb :index
+	end
+
+	get '/signup' do
+		if !session[:id].nil?
+			redirect '/tweets'
+		end
+		erb :'/users/create_user'
+	end
+
+	post '/signup' do
+		@user = User.create(params)
+		if params[:username] == "" || params[:password] == "" || params[:email] == ""
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these checks more easily.